authors/apps/authentication/views.py

```diff
@@ -139,3 +140,27 @@ def post(self, request):
         # serializer.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class VerifyAPIView(APIView):
+    permission_classes = (AllowAny,)
+    serializer_class = VerificationSerializer
+    token_class = TokenGenerator()
+
+    def get(self, request, token):
+        decoded_token = self.token_class.decode_token(token)
+
+        if not isinstance(decoded_token, dict):
+            return Response({'token': 'token is invalid'}, status=status.HTTP_400_BAD_REQUEST)
+        # We call the `serializer.update()` so as to update the data
+        # If serializer.data returns an empty dictionary then user
+        # is successfully registred else return data from the serializer.
+        # Basically, a user can only verify their account once.
+
+        serializer = self.serializer_class(data=decoded_token)
+        serializer.is_valid(raise_exception=True)
+        serializer.verify_user(decoded_token)
+
+        if bool(serializer.data) is False:
+            return HttpResponseRedirect(decoded_token['callbackurl'])
```

Comment:
@ssewilliam, great work here doing a redirect. Are the status codes appended automatically or not. I expected it to have a status code. How are you writing tests for this?

Reply:
@abulojoshua1 
- Redirection automatically returns 3xx status codes, for me, I only need to check if the status  code is 302 when the page redirects.

Reply:
That's great bro! Great staff.